.github/workflows/release.yml

```diff
@@ -0,0 +1,119 @@
+name: Build & Release
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+
```

Comment:
Shall the release be triggered on release/tag creation?
Like this [on release creation](https://github.com/cristianoliveira/funzzy/blob/master/.github/workflows/on-release.yml#L5)

Reply:
My thought was to make a release as easy as possible, by just incrementing the `Cargo.toml` version (and then the workflow deals with tagging).

Hmmm, I'm not sure if workflows trigger from tag pushes by Actions, but if so that would be nice to split it.

Thoughts?

Reply:
Hmm I'm conflicted with this one, on one side, if the artifacts build and upload are connected with the tags, you can create Release notes using the GitHub releases feature ([like this](https://github.com/cristianoliveira/funzzy/releases/tag/v1.0.0)), which is super convenient. On the other, I like the idea of releasing it by changing the version in Cargo.toml, cuz I lost the count of times I released one of my projects just to realize I forgot to bump the version in Cargo.toml and have to do it manually later 😅

Reply:
If you mean the "automatic release notes" (or whatever GitHub calls them), the `softprops/action-gh-release` action has the option `generate_release_notes: true` to automatically add them.

I guess to run tests, it makes sense that it would compile at least once. I feel like we can address that in a separate PR (to try and keep this one related to just releasing). I definitely plan to disable the `on: pullrequest` for this workflow before merging (if Prime agrees).